src/StructuredYson/StructuredYson.tsx

```diff
@@ -259,36 +274,76 @@ export class StructuredYson extends React.PureComponent<Props, State> {
 
         const targetMatchPath = allMatchPaths[nextTotalIndex];
 
-        const collapsedParent = this.findCollapsedParent(targetMatchPath, collapsedState);
+        // Expand all collapsed parents at once
+        let effectiveCollapsedState = collapsedState;
+        let hasCollapsedParents = false;
 
-        // If target is not hidden, it's visible - navigate to it
-        if (collapsedParent === null) {
-            // Count how many visible matches are before our target
-            let visibleMatchCount = 0;
-            for (let i = 0; i < nextTotalIndex; i++) {
-                if (this.findCollapsedParent(allMatchPaths[i], collapsedState) === null) {
-                    visibleMatchCount++;
+        let nextSlash = 0;
+        while ((nextSlash = targetMatchPath.indexOf('/', nextSlash)) !== -1) {
+            const checkPath = targetMatchPath.slice(0, nextSlash);
+            if (collapsedState[checkPath]) {
+                if (!hasCollapsedParents) {
+                    effectiveCollapsedState = {...collapsedState};
+                    hasCollapsedParents = true;
                 }
+                delete effectiveCollapsedState[checkPath];
             }
-
-            // Navigate to the visible match
-            if (visibleMatchCount < matchedRows.length) {
-                this.setState({matchIndex: nextTotalIndex});
-                this.tableRef.current?.scrollToIndex(matchedRows[visibleMatchCount]);
-                this.searchRef.current?.focus();
+            nextSlash++;
+        }
```

Comment:
Is it possible to avoid copy-paste from `findCollapsedParent` here, like:
```ts
const expandCollapsedPath = (path: string) {
    if (effectiveCollapsedState === collapsedState) {
        effectiveCollapsedState = {...collapsedState};
    }
    delete effectiveCollapsedState[collapsedParent];
}

let collapsedParent: string | undefined;
while(collapsedParent = this.findCollapsedParent(..., effectiveCollapsedState)) {
    expandPath(collapsedParent);
}
```
?